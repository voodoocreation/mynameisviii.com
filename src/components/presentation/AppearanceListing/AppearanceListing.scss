@import "../../../scss/variables", "../../../scss/mixins";

.AppearanceListing {
  flex-basis: 100%;
  padding: $spacing-half;

  h3 {
    margin-top: 0;
  }

  a {
    @include panel-2();

    display: flex;
    flex-flow: row-reverse wrap;
    position: relative;
    overflow: hidden;
  }

  &--image {
    position: relative;
    width: 100%;
    overflow: hidden;

    .Loader {
      padding-top: 141.5%;
    }

    &.isRendered {
      display: flex;
      align-items: center;
    }
  }

  &--header::after {
    @include bg-2();

    display: block;
    height: 1px;
    content: "";
    margin-bottom: $spacing;
  }

  &--details {
    position: relative;
    flex-grow: 1;
    padding: $spacing;
  }

  &--status {
    position: absolute;
    top: 50%;
    left: 50%;
    font-size: 30px;
    text-transform: uppercase;
    padding: $spacing-half $spacing;
    border: $text-color 2px dashed;
    border-radius: 10px;
    transform: rotate(-20deg) translate3d(-50%, -50%, 0);
  }

  &--price {
    @include panel-1();

    position: absolute;
    bottom: 0;
    right: 0;
    font-size: 24px;
    padding: $spacing-half $spacing-half $spacing-half ($spacing / 4);

    &::before {
      @include panel-1();
      @include angle-before();

      width: $spacing-double;
    }
  }

  &.isCondensed {
    .AppearanceListing--header::after {
      display: none;
    }
  }

  &.isRendered.isFinished {
    opacity: 0.5;
  }

  &.isRendered:hover {
    opacity: 1;
  }

  @include screen-below($screen-sm-max) {
    &--details {
      padding-bottom: ($spacing * 3);
    }
  }

  @include screen-below($screen-xxs-max) {
    a {
      flex-direction: column-reverse;
    }

    &--details,
    &--image {
      flex-basis: 100%;
    }
  }

  @include screen-above($screen-xs-min) {
    &--details {
      flex-basis: 60%;
    }

    &--image {
      flex-basis: 40%;
    }
  }

  @include screen-above(600px) {
    &--details {
      flex-basis: 70%;
    }

    &--image {
      flex-basis: 30%;
    }

    &--price {
      font-size: 30px;
      padding: $spacing $spacing $spacing $spacing-half;

      &::before {
        width: ($spacing * 3);
      }
    }
  }

  @include screen-above($screen-sm-min) {
    &--details {
      flex-basis: 60%;
    }

    &--image {
      flex-basis: 40%;
    }
  }

  @include screen-above(840px) {
    &--details {
      flex-basis: 65%;
    }

    &--image {
      flex-basis: 35%;
    }
  }

  @include screen-above($screen-lg-min) {
    flex-basis: 50%;
  }

  @include screen-above(1300px) {
    &.isCondensed {
      .AppearanceListing--image {
        position: absolute;
        top: 50%;
        left: 50%;
        width: 250%;
        opacity: 0.2;
        transform: translate3d(-50%, -50%, 0);
        transition: width 200ms ease-out, opacity 200ms ease-out;
      }

      .AppearanceListing--details {
        position: relative;
        z-index: 1;
      }

      a:hover .AppearanceListing--image {
        width: 270%;
        opacity: 0.25;
      }
    }
  }

  @include screen-between(1300px, 1500px) {
    &.isCondensed .AppearanceListing--details {
      font-size: 14px;
    }
  }

  @include screen-between($screen-lg-min, 1678px) {
    &--price {
      font-size: 24px;
      padding: $spacing-half $spacing-half $spacing-half ($spacing / 4);

      &::before {
        width: $spacing-double;
      }
    }
  }

  @include screen-above(1470px) {
    &--details {
      flex-basis: 75%;
    }

    &--image {
      flex-basis: 25%;
    }
  }
}
