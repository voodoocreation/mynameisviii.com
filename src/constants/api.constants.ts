export enum BOOLEAN {
  TRUE = "y",
  FALSE = "n",
}
