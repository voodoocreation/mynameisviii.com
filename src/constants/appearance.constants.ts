export enum STATUS {
  CANCELLED = "EventCancelled",
  POSTPONED = "EventPostponed",
  RESCHEDULED = "EventRescheduled",
  SCHEDULED = "EventScheduled"
}

export enum TYPE {
  ART = "VisualArtsEvent",
  EXHIBITION = "ExhibitionEvent",
  FESTIVAL = "Festival",
  MUSIC = "MusicEvent",
  SCREENING = "ScreeningEvent",
  SOCIAL = "SocialEvent"
}
