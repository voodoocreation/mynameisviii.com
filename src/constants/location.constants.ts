export enum TYPE {
  CITY = "City",
  EVENT_VENUE = "EventVenue",
  MOVIE_THEATER = "MovieTheater",
  MUSIC_VENUE = "MusicVenue",
  PARK = "Park",
  PERFORMING_ARTS_THEATER = "PerformingArtsTheater",
  STADIUM = "StadiumOrArena",
}
