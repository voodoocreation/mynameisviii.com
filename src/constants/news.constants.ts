export enum TYPE {
  APPEARANCE = "appearance",
  NEWS = "news",
  RELEASE = "release"
}
