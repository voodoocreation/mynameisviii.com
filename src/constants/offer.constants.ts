export enum AVAILABILITY {
  IN_STOCK = "InStock",
  IN_STORE = "InStoreOnly",
  LIMITED = "LimitedAvailability",
  OUT_OF_STOCK = "OutOfStock",
  ONLINE = "OnlineOnly",
  PRE_ORDER = "PreOrder",
  PRE_SALE = "PreSale",
  SOLD_OUT = "SoldOut",
}
