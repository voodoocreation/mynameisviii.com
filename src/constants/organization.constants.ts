export enum TYPE {
  CORPORATION = "Corporation",
  CHARITY = "FundingScheme",
  LOCAL_BUSINESS = "LocalBusiness",
  MEDIA = "NewsMediaOrganization",
  PERFORMER = "PerformingGroup",
  PROJECT = "Project",
}
