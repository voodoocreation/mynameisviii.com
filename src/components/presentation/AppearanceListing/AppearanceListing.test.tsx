import dayjs from "dayjs";

import { BOOLEAN } from "../../../constants/api.constants";
import { STATUS } from "../../../constants/appearance.constants";
import {
  appearance,
  location,
  offer,
  organization
} from "../../../models/root.models";
import ComponentTester from "../../../utilities/ComponentTester";
import AppearanceListing from "./AppearanceListing";

const component = new ComponentTester(AppearanceListing).withDefaultProps(
  appearance({
    acts: [
      {
        genre: "Genre",
        imageUrl: "Image URL",
        location: {
          address: "City, Country"
        },
        name: "Act name"
      }
    ],
    description: "Description",
    finishingAt: dayjs()
      .add(4, "day")
      .toISOString(),
    imageUrl: "Image URL",
    isActive: BOOLEAN.TRUE,
    location: {
      latLng: {
        lat: 0,
        lng: 0
      },
      name: "Venue"
    },
    organizer: {
      name: "Organizer"
    },
    startingAt: dayjs()
      .add(3, "day")
      .toISOString(),
    status: STATUS.SCHEDULED
  })
);

describe("[presentation] <AppearanceListing />", () => {
  describe("when only the required props are defined", () => {
    const { wrapper } = component.render();

    it("doesn't have the isCondensed class", () => {
      expect(wrapper.hasClass("isCondensed")).toBe(false);
    });

    it("doesn't render the status notice", () => {
      expect(wrapper.find(".AppearanceListing--status")).toHaveLength(0);
    });

    it("doesn't render the audience meta item", () => {
      expect(wrapper.find(".AppearanceListing--audience")).toHaveLength(0);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when all props are defined", () => {
    const { wrapper } = component
      .withProps({
        audience: "18+",
        images: [
          {
            imageUrl: "Image URL 1",
            title: "Title 1"
          },
          {
            imageUrl: "Image URL 2",
            title: "Title 2"
          }
        ],
        isCondensed: true,
        location: location({
          address: "123 Address",
          name: "Venue"
        }),
        organizer: organization({
          email: "Email",
          logo: "Logo",
          name: "Organizer"
        }),
        rsvpUrl: "RSVP",
        sales: [
          offer({
            name: "Price 1",
            price: 40
          }),
          offer({
            name: "Price 2",
            price: 20
          }),
          offer({
            name: "Price 3",
            price: 25
          })
        ]
      })
      .render();

    it("has the isCondensed class", () => {
      expect(wrapper.hasClass("isCondensed")).toBe(true);
    });

    it("renders the audience meta item", () => {
      expect(wrapper.find(".AppearanceListing--audience")).toHaveLength(1);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when there are no sales", () => {
    const { wrapper } = component
      .withProps({
        sales: []
      })
      .render();

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when there is only one sale", () => {
    const { wrapper } = component
      .withProps({
        sales: [
          offer({
            name: "Price",
            price: 5
          })
        ]
      })
      .render();

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when there are two sales", () => {
    const { wrapper } = component
      .withProps({
        sales: [
          offer({
            name: "Price 1",
            price: 5
          }),
          offer({
            name: "Price 2",
            price: 10
          })
        ]
      })
      .render();

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when there are three sales", () => {
    const { wrapper } = component
      .withProps({
        sales: [
          offer({
            name: "Price 1",
            price: 5
          }),
          offer({
            name: "Price 2",
            price: 10
          }),
          offer({
            name: "Price 2",
            price: 10
          })
        ]
      })
      .render();

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when cancelled", () => {
    const { wrapper } = component
      .withProps({
        sales: [offer()],
        status: STATUS.CANCELLED
      })
      .render();

    it("has the isCancelled class", () => {
      expect(wrapper.hasClass("isCancelled")).toBe(true);
    });

    it("renders the status notice with CANCELLED text", () => {
      expect(wrapper.find(".AppearanceListing--status")).toHaveLength(1);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when postponed", () => {
    const { wrapper } = component
      .withProps({
        sales: [offer()],
        status: STATUS.POSTPONED
      })
      .render();

    it("has the isPostponed class", () => {
      expect(wrapper.hasClass("isPostponed")).toBe(true);
    });

    it("renders the status notice with POSTPONED text", () => {
      expect(wrapper.find(".AppearanceListing--status")).toHaveLength(1);
    });

    it("matches snapshot", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("when the image loads", () => {
    describe("when the onLoad prop isn't defined", () => {
      const { wrapper } = component.mount();

      it("triggers image load", () => {
        wrapper.find("img").simulate("load");
      });

      it("has the isRendered class", () => {
        expect(wrapper.render().hasClass("isRendered")).toBe(true);
      });
    });

    describe("when the onLoad prop is defined", () => {
      const { props, wrapper } = component
        .withProps({
          onLoad: jest.fn()
        })
        .mount();

      it("triggers image load", () => {
        wrapper.find("img").simulate("load");
      });

      it("calls the onLoad prop", () => {
        expect(props.onLoad).toHaveBeenCalled();
      });

      it("has the isRendered class", () => {
        expect(wrapper.render().hasClass("isRendered")).toBe(true);
      });
    });
  });
});
