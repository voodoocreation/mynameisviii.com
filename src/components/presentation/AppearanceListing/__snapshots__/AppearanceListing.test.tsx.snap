// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[presentation] <AppearanceListing /> renders correctly when status=EventCancelled 1`] = `
<article
  class="AppearanceListing isCancelled"
>
  <a
    href="/appearances/undefined"
  >
    <div
      class="AppearanceListing-details"
    >
      <header
        class="AppearanceListing-header"
      >
        <h3 />
      </header>
      <section
        class="AppearanceListing-meta"
      >
        <div
          class="Meta AppearanceListing-status"
        >
           
          <span
            class="Meta-label"
          >
            <span>
              Status
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <span>
              Cancelled
            </span>
          </span>
        </div>
        <div
          class="Meta AppearanceListing-date"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Date
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <time
              class="DateTime"
              datetime="2018-10-10T18:00:00"
              title="October 10, 2018"
            >
              October 10, 2018
            </time>
          </span>
        </div>
        <div
          class="Meta AppearanceListing-time"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Time
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <time
              class="DateTime"
              datetime="2018-10-10T18:00:00"
              title="6:00 PM"
            >
              6:00 PM
            </time>
          </span>
        </div>
        <div
          class="Meta AppearanceListing-location"
          title="location name - undefined"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Location
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            location name
          </span>
        </div>
        <div
          class="Meta AppearanceListing-price"
        >
           
          <span
            class="Meta-label"
          >
            <span>
              Price
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          />
        </div>
      </section>
    </div>
    <figure
      class="AppearanceListing-image"
    >
      <img
        src="imageUrl"
      />
    </figure>
  </a>
</article>
`;

exports[`[presentation] <AppearanceListing /> renders correctly when status=EventPostponed 1`] = `
<article
  class="AppearanceListing isPostponed"
>
  <a
    href="/appearances/undefined"
  >
    <div
      class="AppearanceListing-details"
    >
      <header
        class="AppearanceListing-header"
      >
        <h3 />
      </header>
      <section
        class="AppearanceListing-meta"
      >
        <div
          class="Meta AppearanceListing-status"
        >
           
          <span
            class="Meta-label"
          >
            <span>
              Postponed
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <span>
              Postponed
            </span>
          </span>
        </div>
        <div
          class="Meta AppearanceListing-date"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Date
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <time
              class="DateTime"
              datetime="2018-10-10T18:00:00"
              title="October 10, 2018"
            >
              October 10, 2018
            </time>
          </span>
        </div>
        <div
          class="Meta AppearanceListing-time"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Time
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <time
              class="DateTime"
              datetime="2018-10-10T18:00:00"
              title="6:00 PM"
            >
              6:00 PM
            </time>
          </span>
        </div>
        <div
          class="Meta AppearanceListing-location"
          title="location name - undefined"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Location
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            location name
          </span>
        </div>
        <div
          class="Meta AppearanceListing-price"
        >
           
          <span
            class="Meta-label"
          >
            <span>
              Price
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          />
        </div>
      </section>
    </div>
    <figure
      class="AppearanceListing-image"
    >
      <img
        src="imageUrl"
      />
    </figure>
  </a>
</article>
`;

exports[`[presentation] <AppearanceListing /> renders correctly with all props 1`] = `
<article
  class="AppearanceListing"
>
  <a
    href="/appearances/undefined"
  >
    <div
      class="AppearanceListing-details"
    >
      <header
        class="AppearanceListing-header"
      >
        <h3 />
      </header>
      <section
        class="AppearanceListing-meta"
      >
        <div
          class="Meta AppearanceListing-date"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Date
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <time
              class="DateTime"
              datetime="2018-10-10T18:00:00"
              title="October 10, 2018"
            >
              October 10, 2018
            </time>
          </span>
        </div>
        <div
          class="Meta AppearanceListing-time"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Time
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <time
              class="DateTime"
              datetime="2018-10-10T18:00:00"
              title="6:00 PM"
            >
              6:00 PM
            </time>
          </span>
        </div>
        <div
          class="Meta AppearanceListing-location"
          title="undefined - location address"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Location
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          />
        </div>
        <div
          class="Meta AppearanceListing-audience"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Audience
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            audience
          </span>
        </div>
        <div
          class="Meta AppearanceListing-price"
        >
           
          <span
            class="Meta-label"
          >
            <span>
              Price
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <span
              class="Price PriceRange-min"
            >
              <span>
                NaN
              </span>
            </span>
          </span>
        </div>
      </section>
    </div>
    <figure
      class="AppearanceListing-image"
    >
      <img
        src="imageUrl"
      />
    </figure>
  </a>
</article>
`;

exports[`[presentation] <AppearanceListing /> renders correctly with minimum props 1`] = `
<article
  class="AppearanceListing"
>
  <a
    href="/appearances/undefined"
  >
    <div
      class="AppearanceListing-details"
    >
      <header
        class="AppearanceListing-header"
      >
        <h3 />
      </header>
      <section
        class="AppearanceListing-meta"
      >
        <div
          class="Meta AppearanceListing-date"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Date
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <time
              class="DateTime"
              datetime="2018-10-10T18:00:00"
              title="October 10, 2018"
            >
              October 10, 2018
            </time>
          </span>
        </div>
        <div
          class="Meta AppearanceListing-time"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Time
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <time
              class="DateTime"
              datetime="2018-10-10T18:00:00"
              title="6:00 PM"
            >
              6:00 PM
            </time>
          </span>
        </div>
        <div
          class="Meta AppearanceListing-location"
          title="location name - undefined"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Location
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            location name
          </span>
        </div>
        <div
          class="Meta AppearanceListing-price"
        >
           
          <span
            class="Meta-label"
          >
            <span>
              Price
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          />
        </div>
      </section>
    </div>
    <figure
      class="AppearanceListing-image"
    >
      <img
        src="imageUrl"
      />
    </figure>
  </a>
</article>
`;

exports[`[presentation] <AppearanceListing /> updates \`isRendered\` state after image has loaded 1`] = `
<article
  class="AppearanceListing isRendered"
>
  <a
    href="/appearances/undefined"
  >
    <div
      class="AppearanceListing-details"
    >
      <header
        class="AppearanceListing-header"
      >
        <h3 />
      </header>
      <section
        class="AppearanceListing-meta"
      >
        <div
          class="Meta AppearanceListing-date"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Date
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <time
              class="DateTime"
              datetime="2018-10-10T18:00:00"
              title="October 10, 2018"
            >
              October 10, 2018
            </time>
          </span>
        </div>
        <div
          class="Meta AppearanceListing-time"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Time
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <time
              class="DateTime"
              datetime="2018-10-10T18:00:00"
              title="6:00 PM"
            >
              6:00 PM
            </time>
          </span>
        </div>
        <div
          class="Meta AppearanceListing-location"
          title="location name - undefined"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Location
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            location name
          </span>
        </div>
        <div
          class="Meta AppearanceListing-price"
        >
           
          <span
            class="Meta-label"
          >
            <span>
              Price
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          />
        </div>
      </section>
    </div>
    <figure
      class="AppearanceListing-image"
    >
      <img
        src="imageUrl"
      />
    </figure>
  </a>
</article>
`;

exports[`[presentation] <AppearanceListing /> updates \`isRendered\` state and triggers onLoad prop when image has previously been loaded 1`] = `
<article
  class="AppearanceListing isRendered"
>
  <a
    href="/appearances/undefined"
  >
    <div
      class="AppearanceListing-details"
    >
      <header
        class="AppearanceListing-header"
      >
        <h3 />
      </header>
      <section
        class="AppearanceListing-meta"
      >
        <div
          class="Meta AppearanceListing-date"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Date
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <time
              class="DateTime"
              datetime="2018-10-10T18:00:00"
              title="October 10, 2018"
            >
              October 10, 2018
            </time>
          </span>
        </div>
        <div
          class="Meta AppearanceListing-time"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm.5-13H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Time
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            <time
              class="DateTime"
              datetime="2018-10-10T18:00:00"
              title="6:00 PM"
            >
              6:00 PM
            </time>
          </span>
        </div>
        <div
          class="Meta AppearanceListing-location"
          title="location name - undefined"
        >
          <svg
            attr="[object Object]"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
            />
          </svg>
           
          <span
            class="Meta-label"
          >
            <span>
              Location
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          >
            location name
          </span>
        </div>
        <div
          class="Meta AppearanceListing-price"
        >
           
          <span
            class="Meta-label"
          >
            <span>
              Price
            </span>
            :
          </span>
           
          <span
            class="Meta-value"
          />
        </div>
      </section>
    </div>
    <figure
      class="AppearanceListing-image"
    >
      <img
        src="imageUrl"
      />
    </figure>
  </a>
</article>
`;
